programmer/tinyprog/__init__.py

```diff
@@ -386,6 +467,60 @@ def boot(self):
             # for the boot command data packet.
             pass
 
+    # similar to program_sectors
```

Comment:
This is almost exactly the same as `program_sectors`...there is way too much duplication of code here.